kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstance.java

```diff
@@ -90,25 +100,35 @@ public void updated(Map<String, Object> properties) {
         }
 
         try {
-            ContainerInstanceOptions newProps = new ContainerInstanceOptions(properties);
-
-            if (newProps.getSignatureTrustAnchor().isPresent()) {
-                ValidationResult containerSignatureValidated = validateContainerImageSignature(newProps);
-                String imageDigest = containerSignatureValidated.imageDigest().orElse("?");
-                logger.info("Container signature validation result for {}@{}({}) - {}", newProps.getContainerImage(),
-                        imageDigest, newProps.getContainerImageTag(),
-                        containerSignatureValidated.isSignatureValid() ? "OK" : "FAIL");
-            } else {
-                logger.info("No trust anchor available. Signature validation skipped.");
-            }
 
-            if (newProps.isEnabled()) {
-                this.containerOrchestrationService.registerListener(this);
-            } else {
-                this.containerOrchestrationService.unregisterListener(this);
+            if (this.actualOptions == null || !this.actualOptions.equals(new ContainerInstanceOptions(properties))) {
```

Comment:
As always I would suggest you to use early returns here for improved readability

```suggestion
            if (this.actualOptions != null && this.actualOptions.equals(new ContainerInstanceOptions(properties))) {
                return;
            }
```